simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/Station.java

```diff
@@ -255,4 +267,52 @@ void refreshStateView() {
         this.stationStoreView = state.createView();
     }
 
+    /**
+     * Stops the current connection and establish a new one using
+     * security profile 3.
+     *
+     * @param url endpoint supporting security profile 3
+     */
+    public void switchToSecurityProfile3(String url) {
+        if (isNull(state.getStationCertificate()) || isNull(state.getStationPublicKey()) || isNull(state.getStationPrivateKey())) {
+            throw new IllegalArgumentException("No station certificate found. Please request one!");
+        }
+
+        try {
+            char[] emptyPassword = new char[0];
+
+            PKCS12PfxPdu pfx = SecurityUtils.generateKeyStore(state.getStationCertificate(), state.getStationCertificateChain(), state.getStationPublicKey(), state.getStationPrivateKey());
```

Comment:
Would you mind to consider extracting SSLContext preparations to a class/method?